test/scripts/withdraw.js

```diff
@@ -0,0 +1,254 @@
+      const decimals = tokenData.decimals || 18
+      const { id, token } = await addCustomMintableTokenToExchange(exchange, symbol, decimals, accounts[0])
+      tokenInfo.push({ symbol: symbol, decimals: decimals, id: id, token: token, address: token.address })
+      await token.mint(masterSafe.address, toErc20Units(amount, decimals))
+      const masterBalance = await token.balanceOf(masterSafe.address)
+      assert.equal(fromErc20Units(masterBalance, decimals), amount, "Setup failed, did not deposit amount")
+    }
+
+    return [masterSafe, bracketAddresses, tokenInfo]
+  }
+
+  const evenDeposits = function(bracketAddresses, tokenData, amount) {
+    const numberOfBrackets = bracketAddresses.length
+    const amountToEach = toErc20Units(amount, tokenData.decimals)
+      .div(new BN(numberOfBrackets))
+      .toString()
+    return bracketAddresses.map(bracketAddress => ({
+      amount: amountToEach,
+      tokenAddress: tokenData.address,
+      bracketAddress: bracketAddress,
+    }))
+  }
+
+  const deposit = async function(masterSafe, deposits) {
+    const batchTransaction = await buildTransferApproveDepositFromList(masterSafe.address, deposits)
+    await execTransaction(masterSafe, lw, batchTransaction)
+    // Close auction for deposits to be reflected in exchange balance
+    await waitForNSeconds(301)
+
+    const amountsInExchange = {}
+    const depositsOfBrackets = {}
+    for (const { amount, tokenAddress, bracketAddress } of deposits) {
+      amountsInExchange[tokenAddress] = (amountsInExchange[tokenAddress] || new BN(0)).add(new BN(amount))
+      if (!depositsOfBrackets[tokenAddress]) depositsOfBrackets[tokenAddress] = {}
+      depositsOfBrackets[tokenAddress][bracketAddress] = (depositsOfBrackets[tokenAddress][bracketAddress] || new BN(0)).add(
+        new BN(amount)
+      )
+    }
+
+    for (const { tokenAddress } of deposits) {
+      const token = await ERC20.at(tokenAddress)
+      assert.equal(
+        (await token.balanceOf(exchange.address)).toString(),
+        amountsInExchange[tokenAddress].toString(),
+        "Balance setup failed: the exchange does not hold all tokens"
+      )
+      const tokenDepositsOfBrackets = depositsOfBrackets[tokenAddress]
+      for (const bracketAddress in tokenDepositsOfBrackets) {
+        assert.equal(
+          (await exchange.getBalance(bracketAddress, tokenAddress)).toString(),
+          tokenDepositsOfBrackets[bracketAddress].toString(),
+          "Balance setup failed: the bracket is not the owner of the tokens in the exchange"
+        )
+      }
+    }
+  }
+  it("requests withdrawals", async () => {
+    const amounts = [{ token: { decimals: 18, symbol: "DAI" }, amount: "1000" }]
+    const [masterSafe, bracketAddresses, tokenInfo] = await setup(2, amounts)
+    const deposits = evenDeposits(bracketAddresses, tokenInfo[0], "1000")
+    await deposit(masterSafe, deposits)
+    const depositFile = await tmp.file()
```

Comment:
do we really need the async part of the tmp-promise package? Wouldn't the sync tmp package suffice?

Reply:
What is this new dependency doing for us? Weren't we able to read the deposit file before without the additional dependency?

Reply:
As it is right now, we don't need the async version because I prioritized readability. On the other hand, there is also no reason to avoid the async version, since it doesn't really add any complexity and actually has a more intuitive naming convention IMHO. (For example, in the sync version `.path` is `.name` and `.cleanup()` becomes a separate function `cleanupCallback()`.)